### ISSUE TITLE
Provide detected lockfiles as JSON output

### ISSUE THREAD

**Issue by maxrake** [phylum-dev/cli#982]
# Overview
It would be nice to have a feature that provides the list of lockfiles the CLI is able to detect. The lockfiles are currently automatically detected in commands like `parse` and `analyze` but there is no easy way to extract that list.

* It would be helpful if this feature was exposed as it's own command/subcommand or flag to an existing one
* `phylum-ci` needs to know the lockfiles that are automatically detected
  * It currently does that itself, replicating the logic from the `phylum_lockfile` crate
  * There is a chance of divergence in how these two solutions are implemented

# Acceptance Criteria
- [ ] A list of lockfiles that _would be_ automatically detected by other commands (`parse`, `analyze`, etc.) is provided without having to run those commands
- [ ] The output is available in JSON format
- [ ] Documentation is updated

**Comment by kylewillmon** [comment]
As a related issue: `phylum parse` should include lockfile paths when listing packages.

If we made that change, could `phylum-ci` just use that output from `phylum parse`?

**Comment by cd-work** [comment]
> If we made that change, could phylum-ci just use that output from phylum parse?

I don't think this is realistic. `phylum-ci` should only parse machine-readable json output. But I don't think `phylum parse` should output multiple json objects?

So unless we change the entire `phylum parse` output, it probably won't be in a format suitable for `phylum-ci`?

**Comment by kylewillmon** [comment]
> So unless we change the entire `phylum parse` output, it probably won't be in a format suitable for `phylum-ci`?

I am talking about changing the entire `phylum parse` output. We can discuss ways to make that backwards compatible and/or minimize breakage, but I think `phylum parse` needs to include the information.
